test/tstDetailsRay.cpp

```diff
@@ -0,0 +1,110 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#include <ArborX_Box.hpp>
+#include <ArborX_Ray.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#define BOOST_TEST_MODULE DetailsRay
+
+BOOST_AUTO_TEST_CASE(intersects_box)
+{
+  using ArborX::Box;
+  using ArborX::Point;
+  using ArborX::Ray;
+  constexpr auto &intersects = ArborX::Details::RayTracing::intersects;
+
+  Box unit_box1{{0, 0, 0}, {1, 1, 1}};
+  Box unit_box2{{-1, -1, -1}, {0, 0, 0}};
```

Comment:
What is the value added with the second box?

Reply:
Not much, I can probably test the issue using only one box.
So the issue is when
 
maxCorner[d] - origin[d] 
minCorner[d] - origin[d]

have zero values and then multiply by the inverse of the direction vector. 
It is possible to have 0*inf.